docs: Restructures Zenith quickstart (#6262)
dagger/dagger: docs/versioned_docs/version-zenith/quickstart/292472-arguments.mdx

```shell
dagger -m github.com/shykes/daggerverse/hello call hello --help
```

You should see output that looks like this:

```shell
✔ dagger call hello [0.00s]
┃ Say hello to the world!
┃
┃ Usage:
┃   dagger call hello [flags]
┃
┃ Flags:
┃       --figlet-container Container   Optional container for running the figlet tool
┃       --giant                        Encode the message in giant multi-character letters
┃       --greeting string              An optional greeting (default is "hello")
┃       --name string                  An optional name (default is "world")
┃       --shout                        Make the message uppercase, and add more exclamation
┃                                      points
```

### Pass a string or boolean argument

To pass a string argument to a function, add the corresponding flag to your `dagger call` command, followed by the string value. Passing a boolean argument is even simpler: just add the corresponding flag (the value `true` is implied).

You have already passed a string and boolean argument earlier:

```shell
dagger -m github.com/shykes/daggerverse/hello call test --giant --name daggernaut
```

### Pass a container as argument

> In addition to basic types, Dagger defines powerful core types, which can be used in a function's arguments. For example, with the core `Container` type, a function can literally receive a container as argument. So, a function that builds a container could use a default base image, but allow passing a custom base as an argument; or a custom publishing function could take a container as argument and upload it to a registry.

Reviewer: ```suggestion
Here we start exploring some of the features that make Dagger unique.

In addition to basic types, Dagger defines powerful core types, such as `Container`, which represents the state of an OCI container. This type provides a complete API for building, running and distributing containers. And, like any other type, a function can use `Container` for its arguments.

This means that *a function can receive containers as arguments*. To be clear: this is not a string referencing an image on a remote registry. It is the actual state of a container, managed by the Dagger engine, and passed to a function's code as if it were just another variable. This feature, as far as we know, is unique to Dagger, and opens endless possibilities for assembling complex pipelines where container state flows from function to function - in just a few lines of code.
```

This feature is most powerful when calling a function from your own code (see [the section on creating a module](./428201-custom-modules.mdx)). But you can also pass a container argument from the command-line. To do so, add the corresponding flag, followed by the address of an OCI image. The CLI will dynamically pull the image, and pass the resulting container object as argument to the function.

> To see how this works, call the `hello` function again, but with a custom container in which to execute the `figlet` tool. Make sure that the container has the tool already installed.

Reviewer: ```suggestion
Let's try this with our `hello` function. In order to print giant letters, it dynamically executes the `figlet` utility in a container and returns the container's output. By default it builds a container with figlet installed; but it allows passing a custom container, with the `figletContainer` argument.

Let's try calling `hello` with a custom container, which we know has the `figlet` tool installed:
```

```shell

> dagger -m github.com/shykes/daggerverse/hello call hello --giant --figlet-container index.docker.io/hairyhenderson/figlet

Reviewer: ```suggestion
dagger -m github.com/shykes/daggerverse/hello call hello --greeting bonjour --name daggernaut --giant --figlet-container index.docker.io/hairyhenderson/figlet
```

```

You should see the image being pulled, then the following output:

```shell

> ✔ dagger call hello [1.67s]
> ┃  _   _      _ _                             _     _ _
> ┃ | | | | ___| | | ___    __      _____  _ __| | __| | |
> ┃ | |_| |/ _ \ | |/ _ \   \ \ /\ / / _ \| '__| |/ _` | |
> ┃ |  _  |  __/ | | (_) |   \ V  V / (_) | |  | | (_| |_|
> ┃ |_| |_|\___|_|_|\___( )   \_/\_/ \___/|_|  |_|\__,_(_)
> ┃                     |/

Reviewer: ```suggestion
✔ dagger call hello [1.22s]
┃  _                  _                                                                          
┃ | |__   ___  _ __  (_) ___  _   _ _ __                                                         
┃ | '_ \ / _ \| '_ \ | |/ _ \| | | | '__|                                                        
┃ | |_) | (_) | | | || | (_) | |_| | | _                                                         
┃ |_.__/ \___/|_| |_|/ |\___/ \__,_|_|( )                                                        
┃                  |__/               |/                                                         
┃      _                                               _   _                                     
┃   __| | __ _  __ _  __ _  ___ _ __ _ __   __ _ _   _| |_| |                                    
┃  / _` |/ _` |/ _` |/ _` |/ _ \ '__| '_ \ / _` | | | | __| |                                    
┃ | (_| | (_| | (_| | (_| |  __/ |  | | | | (_| | |_| | |_|_|                                    
┃  \__,_|\__,_|\__, |\__, |\___|_|  |_| |_|\__,_|\__,_|\__(_)                                    
┃              |___/ |___/                                                                       
⧗ 3.20s ✔ 73 ∅ 9
```

```

It worked! What happens if you pass a container that doesn't have the required tool installed?

```shell

> dagger -m github.com/shykes/daggerverse/hello call hello --giant --figlet-container index.docker.io/alpine

Reviewer: ```suggestion
dagger -m github.com/shykes/daggerverse/hello call hello --giant --greeting bonjour --name daggernaut --figlet-container index.docker.io/alpine
```

```

You should see an error telling you that `figlet` could not be found:

```shell
✘ dagger call hello ERROR [0.98s]
[...]
✘ stdout ERROR [0.13s]
✘ exec figlet Hello, world! ERROR [0.11s]
┃ exec: "figlet": executable file not found in $PATH
```

### Pass a directory as argument

Another example is the `Directory` type, representing a directory. To see how this works, use the following command:

```shell
dagger -m github.com/vikram-dagger/daggerverse/fileutils call tree --dir .
```

This example uses a file utility module and calls its `Tree()` function. The `Tree()` function accepts a directory (here, the current working directory) as argument and returns a tree representation of that directory. An abbreviated example of the output is shown below:

```shell
✔ dagger call tree [31.01s]
┃ .
┃ ├── README.md
┃ ├── cli
┃ │   └── git-build
┃ │       ├── README.md
┃ │       └── build.sh
┃ └── sdk
┃     ├── go
┃     │   ├── aws-cdk
┃     │   │   ├── README.md
...
┃ 39 directories, 125 files

> ```

Reviewer: ```suggestion
```

You can also pass a remote git reference, and the Dagger CLI will convert it to a `Directory` type referencing the contents of that repository. For example, let's try listing the source code for the Dagger CLI, from the main branch of the Dagger repository:

```shell
$ dagger -m github.com/vikram-dagger/daggerverse/fileutils call tree --dir https://github.com/dagger/dagger#main:cmd/dagger
```

You should be the same file listing as [this github page](https://github.com/dagger/dagger/tree/main/cmd/dagger):

```shell
✔ dagger call tree [1.15s]
┃ .                                                                                              
┃ ├── call.go                                                                                    
┃ ├── cloud.go                                                                                   
┃ ├── debug.go                                                                                   
┃ ├── engine.go                                                                                  
┃ ├── exec_nonunix.go                                                                            
┃ ├── exec_unix.go                                                                               
┃ ├── flags.go                                                                                   
┃ ├── functions.go                                                                               
┃ ├── gen.go                                                                                     
┃ ├── licenses.go                                                                                
┃ ├── listen.go                                                                                  
┃ ├── log.go                                                                                     
┃ ├── main.go                                                                                    
┃ ├── module.go                                                                                  
┃ ├── module_test.go                                                                             
┃ ├── query.go                                                                                   
┃ ├── run.go                                                                                     
┃ ├── session.go                                                                                 
┃ ├── shell.go                                                                                   
┃ └── version.go                                                                                 
┃                                                                                                
┃ 0 directories, 20 files                                                                        
⧗ 3.09s ✔ 79 ∅ 7
```
